app/src/main/java/chat/rocket/android/helper/UserHelper.kt

```diff
@@ -12,57 +11,37 @@ import javax.inject.Inject
 class UserHelper @Inject constructor(
     private val localRepository: LocalRepository,
     private val getCurrentServerInteractor: GetCurrentServerInteractor,
-    settingsRepository: SettingsRepository
+    private val settingsRepository: SettingsRepository
 ) {
 
-    private val settings: PublicSettings = settingsRepository.get(getCurrentServerInteractor.get()!!)
-
-    /**
-     * Return the display name for the given [user].
-     * If setting 'Use_Real_Name' is true then the real name will be given, or else
-     * the username without the '@' is yielded. The fallback for any case is the username, which
-     * could be null.
-     */
-    fun displayName(user: User): String? {
-        return if (settings.useRealName()) user.name ?: user.username else user.username
-    }
-
-    fun displayName(user: SimpleUser): String {
-        return if (settings.useRealName()) user.name ?: user.username ?: "" else user.username ?: ""
-    }
-
-    /**
-     * Return current logged user's display name.
-     *
-     * @see displayName
-     */
-    fun displayName(): String? {
-        user()?.let {
-            return displayName(it)
-        }
-        return null
-    }
-
     /**
      * Return current logged [User].
      */
-    fun user(): User? {
-        return localRepository.getCurrentUser(serverUrl())
-    }
+    fun user(): User? = getCurrentServerInteractor.get()?.let { localRepository.getCurrentUser(it) }
 
     /**
      * Return the username for the current logged [User].
      */
     fun username(): String? = localRepository.get(LocalRepository.CURRENT_USERNAME_KEY, null)
 
     /**
-     * Whether current [User] is admin on the current server.
+     * Return the display name for the given [user].
+     * If setting 'Use_Real_Name' is true then the real name will be given, otherwise the username
+     * without the '@' is yielded.
      */
-    fun isAdmin(): Boolean {
-        return user()?.roles?.find { it.equals("admin", ignoreCase = true) } != null
+    fun displayName(user: SimpleUser): String {
+        val displayName: String? = getCurrentServerInteractor.get()?.let {
```

Comment:
Can be simplified as
```
    return getCurrentServerInteractor.get()?.let {
        if (settingsRepository.get(it).useRealName()) {
            user.name
        } else {
            user.username
        }
    }.orEmpty()
```
No need for a temporary variable.

Reply:
https://github.com/RocketChat/Rocket.Chat.Android/pull/1570/commits/95ea9692ec0688e87a9ad44a946ad0d7b0a9f7e7